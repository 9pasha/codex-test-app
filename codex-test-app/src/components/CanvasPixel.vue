<template>
  <div class="canvas-px">{{ color }}</div>
</template>

<script>
export default {
  name: "CanvasPixel",
  props: {
    x: Number,
    y: Number
  },
  computed: {
    color() {
      return this.$store.state.pixels[this.y - 1][this.x - 1];
    }
  }
};
</script>

<style scoped>
.canvas-px {
  width: 45px;
  height: 45px;
  border: 1px solid dodgerblue;
  display: flex;
  align-items: center;
  justify-content: center;
}
</style>
