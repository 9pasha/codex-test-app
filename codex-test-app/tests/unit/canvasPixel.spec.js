import { shallowMount, createLocalVue } from "@vue/test-utils";
import CanvasPixel from "@/components/CanvasPixel.vue";
import Vuex from "vuex";

const localVue = createLocalVue();

localVue.use(Vuex);

describe("CanvasPixel.vue", () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        pixels: [
          ["", "x", ""],
          ["o", "x", "o"],
          ["", "o", ""]
        ]
      }
    });
  });
  it("check correct color ''", () => {
    const wrapper = shallowMount(CanvasPixel, {
      localVue,
      store,
      propsData: { x: 1, y: 1 }
    });

    expect(wrapper.is(CanvasPixel)).toBeTruthy();
    expect(wrapper.vm.x).toEqual(1);
    expect(wrapper.vm.y).toEqual(1);
    expect(wrapper.text()).toMatch("");
  });
  it("check correct color 'o'", () => {
    const wrapper = shallowMount(CanvasPixel, {
      localVue,
      store,
      propsData: { x: 2, y: 3 }
    });

    expect(wrapper.is(CanvasPixel)).toBeTruthy();
    expect(wrapper.vm.x).toEqual(2);
    expect(wrapper.vm.y).toEqual(3);
    expect(wrapper.text()).toMatch("o");
  });
  it("check correct color 'x'", () => {
    const wrapper = shallowMount(CanvasPixel, {
      localVue,
      store,
      propsData: { x: 2, y: 2 }
    });

    expect(wrapper.is(CanvasPixel)).toBeTruthy();
    expect(wrapper.vm.x).toEqual(2);
    expect(wrapper.vm.y).toEqual(2);
    expect(wrapper.text()).toMatch("x");
  });
  it("check render", () => {
    const wrapper = shallowMount(CanvasPixel, {
      localVue,
      store,
      propsData: { x: 3, y: 1 }
    });

    expect(wrapper.html()).toContain(`<div class="canvas-px"></div>`);
    expect(wrapper.text()).toEqual("");
  });
  it("should match snapshots", () => {
    const wrapper = shallowMount(CanvasPixel, {
      localVue,
      store,
      propsData: { x: 2, y: 2 }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
