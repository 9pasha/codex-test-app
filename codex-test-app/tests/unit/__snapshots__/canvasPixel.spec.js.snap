// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanvasPixel.vue should match snapshots 1`] = `<div class="canvas-px">x</div>`;
